contracts/deploy/DataFeedRouter/DataFeedRouter.cjs

```diff
@@ -0,0 +1,90 @@
+const path = require('node:path')
+const {
+  loadJson,
+  loadMigration,
+  updateMigration,
+  loadDeployments
+} = require('../../scripts/v0.1/utils.cjs')
+
+const func = async function (hre) {
+  console.log('DataFeedRouter')
+
+  const { deployments, getNamedAccounts, network } = hre
+  const { deploy } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  const migrationDirPath = `./migration/${network.name}/DataFeedRouter`
+  const migrationFilesNames = await loadMigration(migrationDirPath)
+
+  for (const migration of migrationFilesNames) {
+    const config = await loadJson(path.join(migrationDirPath, migration))
+    let dataFeedRouter = undefined
+
+    if (config.deploy) {
+      console.log('deploy')
+
+      const dataFeedRouterDeployment = await deploy('DataFeedRouter', {
+        args: [],
+        from: deployer,
+        log: true
+      })
+
+      dataFeedRouter = await ethers.getContractAt(
+        'DataFeedRouter',
+        dataFeedRouterDeployment.address
+      )
+
+      console.log('DataFeedRouter:', dataFeedRouterDeployment)
+    }
+
+    dataFeedRouter = dataFeedRouter
+      ? dataFeedRouter
+      : await ethers.getContractAt('DataFeedRouter', config.dataFeedRouterAddress)
+
+    if (config.updateProxies) {
```

Comment:
It looks like `updateAll` and `proxyList` are exclusive, and the only difference is what kind of feeds we update. I think this would be a bit clearer.

```typescript
  if (config.updateProxies) {
      console.log('update proxies')
      const updateProxiesConfig = config.updateProxies

      const feedNames = []
      const addresses = []

      if (updateProxiesConfig.updateAll) {
        const deployments = await loadDeployments(`./deployments/${network.name}`)

        for (const key in deployments) {
          if (key.includes('AggregatorProxy')) {
            const feedName = key.split('_')[1]
            const address = deployments[key]
            feedNames.push(feedName)
            addresses.push(address)
          }
        }
      } else {
        for (const proxyObject of proxyList) {
          const feedName = proxyObject.feedName
          const address = proxyObject.name
          feedNames.push(feedName)
          addresses.push(address)
        }
      }

      if (feedNames.length == 0 || addresses.length == 0) {
        throw new Error('no proxies to update')
      }
      const tx = await (await dataFeedRouter.updateProxyBulk(feedNames, addresses)).wait()
      console.log(`bulk inserted feeds: ${tx.events[0].args[0]}`)
      console.log(`bulk inserted addresses:${tx.events[0].args[1]}`)
  }
```